src/database/database-sqlite3.cpp

```diff
@@ -606,3 +606,140 @@ void PlayerDatabaseSQLite3::listPlayers(std::vector<std::string> &res)
 
 	sqlite3_reset(m_stmt_player_list);
 }
+
+/*
+ * Auth database
+ */
+
+AuthDatabaseSQLite3::AuthDatabaseSQLite3(const std::string &savedir) :
+		Database_SQLite3(savedir, "auth"), AuthDatabase()
+{
+}
+
+AuthDatabaseSQLite3::~AuthDatabaseSQLite3()
+{
+	FINALIZE_STATEMENT(m_stmt_read)
+	FINALIZE_STATEMENT(m_stmt_write)
+	FINALIZE_STATEMENT(m_stmt_create)
+	FINALIZE_STATEMENT(m_stmt_delete)
+	FINALIZE_STATEMENT(m_stmt_list_names)
+}
+
+void AuthDatabaseSQLite3::createDatabase()
+{
+	assert(m_database); // Pre-condition
+
+	SQLOK(sqlite3_exec(m_database,
+			      "CREATE TABLE IF NOT EXISTS `auth` ("
+			      "`id` INTEGER PRIMARY KEY AUTOINCREMENT,"
+			      "`name` VARCHAR(32),"
```

Comment:
please add an UNIQUE index on the name field as it's unique, this will prvent the ugly limit 1 suggestion

Reply:
Shouldn't it be possible to migrate still?

Reply:
Not completely. The code is currently a drop-in for the `sauth` mod, in that it can use the same database file which already contains all the players. But in that case, the `CREATE` statement will not get run, and the DB will continue to not have the unique key in its table definition for all eternity.

Reply:
having compliant data with sauth is okay.
Scheme: no, please do the right scheme, it's the core engine, we shouldn't be 100% compliant with mod with some design problems

Reply:
ID is nice, it's better than my player implementation which is lazy and use a string as a primary which is bad in terms of performance for huge db (MT players & auth will not be)

Reply:
Please also add a dbversion table like in map db permitting to track changes and upgrade the scheme

Reply:
I'm sorry, I can't find what you mean with a dbversion table. Can you give a link in GitHub somewhere?